derive-impl/src/compile_bytecode.rs

```diff
@@ -369,12 +374,11 @@ pub fn impl_py_compile(
         .source
         .compile_single(args.mode, args.module_name, compiler)?;
 
-    let bytes = code.to_bytes();
-    let bytes = LitByteStr::new(&bytes, Span::call_site());
+    let frozen = frozen_lib::FrozenCodeObject::encode(&code);
+    let bytes = LitByteStr::new(&frozen.bytes, Span::call_site());
 
     let output = quote! {
-        #crate_name::CodeObject::from_bytes(#bytes)
-            .expect("Deserializing CodeObject failed")
+        #crate_name::frozen_lib::FrozenCodeObject { bytes: &#bytes[..] }
```

Comment:
I think the name `py_compile` doesn't imply its result will be compressed but getting code object directly.
Renaming or making a new macro?

Reply:
Perhaps, yeah, but because `vm.ctx.new_code()` still accepts it, it won't actually make a difference for users.

Reply:
They already were - the old CodeObject::from_bytes also did lz4 decompression, and py_compile!() just generated CodeObject::from_bytes(b"..."). I don't think `compile` returning a frozen code object is necessarily unintuitive (they're expecting it to compile it at build time and embed it in the binary; what's that if not freezing?) and again, since it's still accepted in the same places as before it doesn't actually make much of a difference to the user